base/compiler/abstractinterpretation.jl

```diff
@@ -1566,6 +1566,7 @@ function abstract_apply(interp::AbstractInterpreter, argtypes::Vector{Any}, si::
             if i != length(ctypes)
                 # No point carrying forward the info, we're not gonna inline it anyway
                 retinfo = NoCallInfo()
+                effects = Effects()
             end
```

Comment:
Also `rt = Any`? or is that part of the contractual requirement of `bail_out_apply`

Reply:
Yeah `rt === Any` holds true usually. But since we allow overloading of `bail_out_apply` we should make sure that we widen `rt` also. I will change this PR so that it sorts out the correctness of the bail out interface.